New properties: fill-opacity and fill-rule (#35007)
mdn/content: files/en-us/web/css/fill-opacity/index.md

> [!NOTE]
> The `fill-opacity` property only applies to {{SVGElement('circle')}}, {{SVGElement('ellipse')}}, {{SVGElement('path')}}, {{SVGElement('polygon')}}, {{SVGElement('polyline')}}, {{SVGElement('rect')}}, {{SVGElement('text')}}, {{SVGElement('textPath')}}, and {{SVGElement('tspan')}} elements nested in an {{SVGElement("svg")}}. It doesn't apply other SVG, HTML, or pseudo-elements.

## Syntax

```css
/* numeric and percentage values */
fill-opacity: 0.2;
fill-opacity: 20%;

/* Global values */
fill-opacity: inherit;
fill-opacity: initial;
fill-opacity: revert;
fill-opacity: revert-layer;
fill-opacity: unset;
```

### Values

The {{cssxref("number")}} and {{cssxref("percentage")}} values denotes the opacity of the `fill` of the element.

- {{cssxref("number")}}

  - : A numeric value between `0` and `1`, inclusive.

- {{cssxref("percentage")}}

  - : A percentage value between `0%` and `100%`, inclusive.

> With `0` or `0%`, the element is fully transparent (that is, invisible). With `1` or `100%`, the element is fully opaque. With values in between, the element is translucent, with any content behind the element being visible.

Reviewer: Maybe we can drop "(that is, invisible)" or do you want to keep that in? Minor reword if you think it's an improvement

```suggestion
With values of `0` or `0%`, the element is fully transparent and with values of `1` or `100%`, the element is fully opaque. With values in between, the element is semi-transparent, with any content behind the element being visible.
```

## Formal definition

> {{CSSInfo}}

Reviewer: This is not in mdn/data repo, so it looks broken

```css hidden
svg {
  border: 1px solid;
  height: calc(100vh - 20px);
  background: repeating-linear-gradient(
    to bottom right,
    transparent 0 10px,
    #ccc 10px 11px
  );
}
```

```css
svg > * {
  fill: black;
}
rect:last-of-type {
  fill-opacity: 0.1;
}
circle {
  fill-opacity: 0.6;
}
ellipse {
  fill-opacity: 40%;
}
line {
  fill-opacity: 10%;
}
polyline {
  fill-opacity: 50%;
}
path {
  fill-opacity: 0.5;
}

circle,
ellipse {
  stroke: black;
  stroke-width: 3px;
}
```

#### Results

{{EmbedLiveSample("Defining the fill opacity of SVG elements", "300", "360")}}

Only two elements are fully opaque: the first rectangle and the line. The first rectangle is not matched by any of the selectors, therefore no CSS is applied and the `fill` is fully opaque. The `line` is matched, with `fill-opacity: 10%` set. However, the line has no `fill` paint operation— rather, only the `stroke` is visible—therefore the declaration has no effect.

## Specifications

{{Specifications}}

## Browser compatibility

{{Compat}}

## See also

- SVG {{SVGAttr("fill-opacity")}} attribute

> - Presentation properties: `fill-opacity`, {{cssxref("clip-rule")}}, {{cssxref("color-interpolation-filters")}}, {{cssxref("fill-rule")}}, {{cssxref("fill")}}, {{cssxref("marker-end")}}, {{cssxref("marker-mid")}}, {{cssxref("marker-start")}}, {{cssxref("shape-rendering")}}, {{cssxref("stop-color")}}, {{cssxref("stop-opacity")}}, {{cssxref("stroke")}}, {{cssxref("stroke-dasharray")}}, {{cssxref("stroke-dashoffset")}}, {{cssxref("stroke-linecap")}}, {{cssxref("stroke-linejoin")}}, {{cssxref("stroke-miterlimit")}}, {{cssxref("stroke-opacity")}}, {{cssxref("stroke-width")}}, {{cssxref("text-anchor")}}, and {{cssxref("vector-effect")}}

Reviewer: There's a lot of flaws reported (20 per doc) saying "wrong macro", e.g., `{{cssxref("fill")}}` - is this something you think Yari should __not__ be flagging as a flaw?

Author: we are creating a bunch of new pages this week and next; all these missing pages will be created and we will stop the redirect.

Reviewer: okie doke

- {{cssxref("opacity")}}
- {{cssxref("background-color")}}

> - {{cssxref("color_value", "&lt;color>")}}

Reviewer: just noticed we're mixing `&lt;` and `>` here

Author: In HTML, the `<` needs to be escaped (otherwise it's an element). the `>` doesn't need to be escaped..
